valory-xyz/open-aea
# Add p2p* packages

## Proposed changes

### Porting plan / steps:

0. ensure tests pass on `agents-aea`
	test_connections/test_p2p_libp2p* -> **yes**
	tests/test_packages/test_protocols/test_acn.py -> **yes**

1. copy directly into the same location in open-aea: 
    packages/fetchai/connections/p2p_libp2p/
    packages/fetchai/connections/p2p_libp2p_client/
    packages/fetchai/connections/p2p_libp2p_mailbox/
    packages/fetchai/protocols/acn/
    tests/test_packages/test_connections/test_p2p_libp2p/
    tests/test_packages/test_connections/test_p2p_libp2p_client/
    tests/test_packages/test_connections/test_p2p_libp2p_mailbox/
    tests/test_packages/test_protocols/test_acn.py

2. port missing functions and configurations
	modified:   tests/conftest.py

3. ensure test passing on open-aea side
	test_connections/test_p2p_libp2p* -> **no**
	tests/test_packages/test_protocols/test_acn.py -> **yes**

4. once they do we can change other things (e.g. author, path, etc.)

### Fixing failing tests:

	tests/test_packages/test_protocols/test_acn.py -> yes

	test_connections/test_p2p_libp2p/ -> no
		-rw-rw-r--  1 karrenbelt karrenbelt  9205 Feb 14 22:48 test_aea_cli.py 			-> fail
		-rw-rw-r--  1 karrenbelt karrenbelt  3341 Feb 14 22:48 test_build.py 			-> pass
		-rw-rw-r--  1 karrenbelt karrenbelt 32440 Feb 14 22:48 test_communication.py 	-> fail
		-rw-rw-r--  1 karrenbelt karrenbelt 15795 Feb 14 22:48 test_errors.py 			-> fail
		-rw-rw-r--  1 karrenbelt karrenbelt 20450 Feb 14 22:48 test_fault_tolerance.py 	-> fail
		-rw-rw-r--  1 karrenbelt karrenbelt  9595 Feb 14 22:48 test_integration.py 		-> fail
		-rw-rw-r--  1 karrenbelt karrenbelt 23019 Feb 14 22:48 test_public_dht.py 		-> fail
		-rw-rw-r--  1 karrenbelt karrenbelt  5243 Feb 14 22:48 test_slow_queue.py 		-> fail



types of errors are connection and private key errors. Will try to resolve the connection errors first

- ```Exception: Couldn't connect to libp2p process within timeout```
- ```aea.mail.base.AEAConnectionError: Failed to connect the multiplexer: Error: Exception("Couldn't connect to libp2p process within timeout")```
- ```click.exceptions.ClickException: Cannot find connection private key with id 'fetchai'. Connection 'fetchai/p2p_libp2p:0.26.0' requires this. Please use `aea generate-key fetchai connection_fetchai_private_key.txt` and `aea add-key fetchai connection_fetchai_private_key.txt --connection` to add a connection private key with id 'fetchai'.```


Will branch again and start another PR on top of this PR to attempt to resolve the errors with the tests


## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_


## Checklist

_Put an `x` in the boxes that apply._

- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have checked that code coverage does not decrease.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...
